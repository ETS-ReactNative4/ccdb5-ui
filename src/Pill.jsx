import React from 'react';
import { connect } from 'react-redux';
import { removeFilter } from './actions/filter'
import { SLUG_SEPARATOR } from './constants'
import './Pill.less';

export const Pill = ({fieldName, value, trimmed, remove}) => {
  return (
    <li className="pill flex-fixed">
      <span className="name">{ trimmed }</span>
      <button onClick={ remove } 
              title={'Remove ' + trimmed + ' as a filter'}>
          <span className="cf-icon cf-icon-delete"></span>
      </button>
    </li>
  );
}

export const mapStateToProps = (state, ownProps) => {
  const parts = ownProps.value.split(SLUG_SEPARATOR)
  const trimmed = parts.length > 1 ? parts.pop() : parts[0];

  return {
    ...ownProps,
    trimmed
  }
}

export const mapDispatchToProps = (dispatch, props) => {
  return {
    remove: () => { dispatch(removeFilter(props.fieldName, props.value))}
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Pill);