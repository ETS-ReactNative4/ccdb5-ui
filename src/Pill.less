@import (less) "base.less";

.pill {
    padding:  @grid_gutter-width/3 25px @grid_gutter-width/3 @grid_gutter-width/3;
    border: 1px solid @teal-40;
    margin-right: 4px;
    border-radius: .28571429em;
    background-color: @teal-20;
    font-weight: 500;
    color: @black;
    font-face: webfont-regular();
    font-size: @size-v;

    // Close button
    button {
    padding: 0;
    padding-left: 5px;
    border: 0;
    border-left: 1px solid @gray-20;
    margin: 0;
    background-color: transparent;
    color: @teal-80;
    position: absolute;
    right: 4px;
    top: 50%;
    transform: translateY(-50%);
      &:hover,
      &:active {
        background-color: @gray-40;
      }
    }
}
