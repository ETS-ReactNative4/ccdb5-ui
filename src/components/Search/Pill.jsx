import './Pill.less'
import { filterPatch, SLUG_SEPARATOR } from '../../constants'
import { removeFilter, replaceFilters } from '../../actions/filter'
import { coalesce } from '../../utils'
import { connect } from 'react-redux'
import { getUpdatedFilters } from '../../utils/filters'
import iconMap from '../iconMap'
import PropTypes from 'prop-types'
import React from 'react'

export class Pill extends React.Component {
  render() {
    const { value, remove } = this.props
    const trimmed = value.split( SLUG_SEPARATOR ).pop()
    return (
      <li className="pill flex-fixed">
        <span className="name">{ trimmed }</span>
        <button onClick={ () => remove( this.props ) }>
          <span className="u-visually-hidden">
            { `Remove ${ trimmed } as a filter` }
          </span>
          { iconMap.getIcon( 'delete' ) }
        </button>
      </li>
    )
  }
}
export const mapStateToProps = ( state, ownProps ) => {
  const aggs = coalesce( state.aggs, ownProps.fieldName, [] )
  const filters = coalesce( state.query, ownProps.fieldName, [] )
  return {
    aggs,
    filters
  }
}

export const mapDispatchToProps = ( dispatch, ownProps ) => ( {
  remove: props => {
    const filterName = props.value
    const { fieldName } = ownProps
    if ( filterPatch.includes( fieldName ) ) {
      const updatedFilters = getUpdatedFilters( filterName, props.filters,
        props.aggs, fieldName )
      dispatch( replaceFilters( fieldName, updatedFilters ) )
    } else {
      dispatch( removeFilter( fieldName, filterName ) )
    }
  }
} )

export default connect( mapStateToProps, mapDispatchToProps )( Pill );

Pill.propTypes = {
  fieldName: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired
}
