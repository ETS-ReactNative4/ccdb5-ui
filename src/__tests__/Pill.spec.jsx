import React from 'react';
import { Pill } from '../Pill';
import renderer from 'react-test-renderer';

describe('component:Pill', () => {
  it('renders without crashing', () => {
    const target = renderer.create(
      <Pill fieldName="foo" value="bar" />
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
