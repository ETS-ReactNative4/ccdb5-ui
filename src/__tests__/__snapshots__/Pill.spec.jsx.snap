exports[`component:Pill renders without crashing 1`] = `
<li
  className="pill flex-fixed">
  <span
    className="name">
    bar
  </span>
  <button
    onClick={undefined}
    title="Remove bar as a filter">
    <span
      className="cf-icon cf-icon-delete" />
  </button>
</li>
`;
